endpoints/api/logs.py

```diff
@@ -440,8 +477,28 @@ def post(self, orgname, parsed_args):
             end_time = parsed_args["endtime"]
 
             export_id = _queue_logs_export(start_time, end_time, request.get_json(), orgname)
+
+            log_action(
+                "export_logs_success",
+                orgname,
+                {
+                    "export_id": export_id,
+                    "url": request.get_json().get("callback_url") or None,
+                    "email": request.get_json().get("callback_email") or None,
+                },
+            )
+
             return {
                 "export_id": export_id,
             }
 
+        log_action(
+            "export_logs_failure",
+            orgname,
+            {
+                "error": "unauthorized",
+                "url": request.get_json().get("callback_url") or None,
+                "email": request.get_json().get("callback_email") or None,
+            },
+        )
```

Comment:
I think instead of logging failures at the api call level, we should log them if the actual background job fails.

Reply:
I don't think that's possible. I did add some additional logic that would grab some exceptions that the API may have. But what you're saying is out of the API scope. Exporting of logs is an async operation, the worker does the job and it picks it up from `queueitem` table. The API is only here to tell the worker that some work needs to be done. If the worker generates an exception and cannot continue, the worker should send a message to the initial e-mail or URL and inform the client that failure happened. I think it does that today. I'll change the wording in the audit logs to reflect that: not `successful` but `queued for execution`.